src/ripple/basics/tagged_integer.h

```diff
@@ -157,6 +163,19 @@ class tagged_integer
         return tagged_integer (lhs.m_value - rhs);
     }
 
+    template <class OtherInt>
+    friend
+    typename std::enable_if <
+        std::is_integral <OtherInt>::value &&
+            sizeof (OtherInt) <= sizeof (Int),
```

Comment:
I think that being able to control the operations available would be good. Rationale: Consider `LedgerIndex`: it makes no sense to multiply or divide ledger indices or to do bitwise operations on them, so if possible those operations, if attempted, ought to trigger compiler errors. On the other hand, it makes sense to perform a modulo. So it's a bit of tricky to strike the right balance.

I'm fine with limiting the scope to `tagged_integer<Int, sometag>` not silently interoperating with other integers, native or tagged.